azurelinuxagent/common/osutil/default.py

```diff
@@ -129,24 +133,40 @@ def get_firewall_will_wait(self):
                 else ""
         return wait
 
-    def remove_firewall(self):
+    def _delete_rule(self, rule):
+        """
+        Continually execute the delete operation until the return
+        code is non-zero.
+        """
+        while True:
```

Comment:
`while True` makes me nervous; can we put in another escape hatch?

Reply:
Give me some suggestions.  What do you typically do?

Reply:
Count or time, so keep removing rules until done or we hit some large number, for example.